reason_4.0/lib/core/minisite_templates/modules/events.php

```diff
@@ -3573,17 +3574,27 @@ function show_list_export_links()
 		}
 		
 		$query_string = $this->construct_link(array('start_date'=>$start_date,'view'=>'','end_date'=>'','format'=>'ical'));
+		$calendar_url = REASON_HOST.$this->parent->pages->get_full_url( $this->page_id );
+		$webcal_url = 'webcal://'.$calendar_url.$query_string;
+		$gcal_url = 'https://calendar.google.com/calendar/render?cid='.str_replace('&amp;','%26',urlencode($webcal_url));
```

Comment:
Seems to me that `EventsModule::construct_link()` produces an already-html-encoded string -- which is useful when you are producing HTML, but in this case you aren't!

I'd add a third parameter to `EventsModule::construct_link()`, `$html_encode = true`, and then either use `&amp;` or just `&` in construct_link() based on what is passed.

Then in your new code you should be able to simply call construct_link($array, true, false) to get a non-html-encoded query string, and not have to do the `str_replace()` here.